module-apps/apps-common/widgets/DateSetSpinner.cpp

```diff
@@ -0,0 +1,267 @@
+// Copyright (c) 2017-2022, Mudita Sp. z.o.o. All rights reserved.
+// For licensing, see https://github.com/mudita/MuditaOS/LICENSE.md
+
+#include "DateSetSpinner.hpp"
+
+#include "BellSideListItemStyle.hpp"
+#include <FontManager.hpp>
+#include <RawFont.hpp>
+#include <apps-common/widgets/BellBaseLayout.hpp>
+#include <gui/widgets/ImageBox.hpp>
+#include <gui/widgets/text/Label.hpp>
+#include <time/time_locale.hpp>
+
+namespace gui
+{
+    namespace date_set_spinner
+    {
+        namespace focus
+        {
+            constexpr auto size = 4U;
+        } // namespace focus
+
+        constexpr auto focusFontName    = style::window::font::large;
+        constexpr auto noFocusFontName  = style::window::font::largelight;
+        constexpr std::uint8_t step     = 1;
+        constexpr std::uint8_t dayMin   = 1;
+        constexpr std::uint8_t dayMax   = 31;
+        constexpr std::uint8_t monthMin = 1;
+        constexpr std::uint8_t monthMax = 12;
+
+    } // namespace date_set_spinner
+
+    using namespace date_set_spinner;
+
+    DateSetSpinner::DateSetSpinner(Item *parent, VBox *titleBox, Length x, Length y, Length w, Length h)
+        : HBox(parent, x, y, w, h)
+    {
+        setAlignment(Alignment(Alignment::Horizontal::Center, Alignment::Vertical::Center));
+        setEdges(RectangleEdge::None);
+
+        title = new TextFixedSize(titleBox);
+        title->setMaximumSize(style::bell_base_layout::w, style::bell_base_layout::outer_layouts_h);
+        title->setFont(style::bell_sidelist_item::title_font);
+        title->setEdges(gui::RectangleEdge::None);
+        title->activeItem = false;
+        title->setAlignment(Alignment(gui::Alignment::Horizontal::Center, gui::Alignment::Vertical::Center));
+        title->setRichText(utils::translate("app_settings_title_day"));
+        title->drawUnderline(false);
+
+        attachDateField(day, U8IntegerSpinnerFixed::range{dayMin, dayMax, step});
+        attachSlash(firstSlash);
+        attachDateField(month, U8IntegerSpinnerFixed::range{monthMin, monthMax, step});
+        attachSlash(secondSlash);
+        attachDateField(year,
+                        U16IntegerSpinnerFixed::range{utils::time::Locale::min_years,
+                                                      utils::time::Locale::max_years,
+                                                      static_cast<std::uint16_t>(step)});
+
+        resizeItems();
+        setFont();
+
+        focusChangedCallback = [&](Item &) {
+            if (editMode != EditMode::Edit) {
+                setFocusItem(nullptr);
+                return false;
+            }
+
+            if (lastFocus != nullptr) {
+                updateFocus(lastFocus);
```

Comment:
here you pass `lastFocus` as a parameter, but in DateSetSpinner.cpp:235 you set it to `lastFocus`, so basically 
```C
lastFocus = lastFocus?
```

Reply:
I did copy this part from TimeSetSpinner. In this particular call It would be a self assigment as you have mentioned. But is it a problem here?

Reply:
In this case it's not a problem. Apart from that I simplified focus update logic a bit ;)